src/basic/process-util.c

```diff
@@ -129,6 +129,13 @@ int get_process_cmdline(pid_t pid, size_t max_length, bool comm_fallback, char *
 
         (void) __fsetlocking(f, FSETLOCKING_BYCALLER);
 
+        if (max_length == 0) {
+                /* This is supposed to be a safety guard against runaway command lines. */
+                long l = sysconf(_SC_ARG_MAX);
+                assert(l > 0);
```

Comment:
Note that the `assert` will probably be removed in production builds so if `sysconf(_SC_ARG_MAX)` would return `-1` (in case the symbolic constant doesn't exist) the `max_length` would probably end up having a maximum possible `size_t` value.

It is probably not good to assume that the particular `sysconf` never fails on some machine, so maybe this should be changed to an if?

Reply:
`sysconf` should not return -1. `_SC_ARG_MAX` is obviously a known constant. This could only happen if somebody tries to run this with a different libc than the common ones. In that case, I hope they compile at least once with asserts enabled and run the tests ;)